src/snyk/common/services/learnService.ts

```diff
@@ -0,0 +1,113 @@
+import axios from 'axios';
+import { isCodeIssue, isOssIssue, OpenCommandIssueType } from '../../common/commands/types';
+import type { completeFileSuggestionType } from '../../snykCode/interfaces';
+import { OssIssueCommandArg } from '../../snykOss/views/ossVulnerabilityTreeProvider';
+import { ILog } from '../logger/interfaces';
+
+const ecosystems = [
+  'all',
+  'java',
+  'javascript',
+  'python',
+  'golang',
+  'php',
+  'cpp',
+  'csharp',
+  'kubernetes',
+  'ruby',
+  'elixir',
+  'docker',
+  'npm',
+] as const;
+type LessonEcosystem = typeof ecosystems[number];
+
+const isValidEcosystem = (potentialEcosystem: string | null | undefined): potentialEcosystem is LessonEcosystem => {
+  return ecosystems.includes(potentialEcosystem as LessonEcosystem);
+};
+
+export type Lesson = {
+  title: string;
+  lessonId: string;
+  ecosystem: LessonEcosystem;
+  url: string;
+};
+
+type PackageMapper = {
+  [key in LessonEcosystem]?: string[];
+};
+
+export interface ILearnService {
+  getLesson(options: { ecosystem: LessonEcosystem; cwes?: string[]; rules?: string[] }): Promise<Lesson | null>;
+}
+
+export class LearnService implements ILearnService {
+  private ecosystem: LessonEcosystem;
+  private cwes: string[] = [];
+
+  constructor(
+    private readonly issue: OssIssueCommandArg | completeFileSuggestionType,
+    private readonly issueType: OpenCommandIssueType,
+    private readonly logger: ILog,
+  ) {
+    this.logger = logger;
+    if (isCodeIssue(issue, issueType)) {
+      this.cwes = issue?.cwe || [];
+      this.ecosystem = LearnService.convertCodeIdToEcosystem(issue.id);
+    } else if (isOssIssue(issue, issueType)) {
+      this.cwes = issue?.identifiers?.CWE || [];
+      this.ecosystem = LearnService.convertOSSProjectTypeToEcosystem(issue.packageManager);
+    } else {
+      logger.error(`Issue type "${issueType}" not supported`);
+    }
+  }
+
+  static convertCodeIdToEcosystem(issueId: string): LessonEcosystem {
+    const ecosystem = (issueId || '').split(/\/|%2F/g)[0]; // %2F is the url encoding for /
```

Comment:
When there is no issueId, and it's going to call `split()` on an empty string, this function should always return 'all', right?
If so, I would suggest to add  `if (!issueId) return 'all';` as first line of this function;

Reply:
Yep can do that